Remotion/Web/UnitTests/Core/Context/HttpContextStorageProviderWithAsyncLocalBasedFallbackTest.cs

```diff
@@ -0,0 +1,161 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// The re-motion Core Framework is free software; you can redistribute it 
+// and/or modify it under the terms of the GNU Lesser General Public License 
+// as published by the Free Software Foundation; either version 2.1 of the 
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful, 
+// but WITHOUT ANY WARRANTY; without even the implied warranty of 
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the 
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+// 
+using System;
+using System.Collections;
+using System.Reflection;
+using System.Threading;
+using System.Web;
+using NUnit.Framework;
+using Remotion.Development.Web.UnitTesting.AspNetFramework;
+using Remotion.Web.Context;
+
+namespace Remotion.Web.UnitTests.Core.Context
+{
+  [TestFixture]
+  public class HttpContextStorageProviderWithAsyncLocalBasedFallbackTest
+  {
+    private HttpContextStorageProviderWithAsyncLocalBasedFallback _provider;
+    private HttpContext _testContext;
+
+    [SetUp]
+    public void SetUp ()
+    {
+      _testContext = HttpContextHelper.CreateHttpContext("x", "y", "z");
+      HttpContext.Current = _testContext;
+      _provider = new HttpContextStorageProviderWithAsyncLocalBasedFallback();
+    }
+
```

Comment:
please add a teardown with `HttpContext.Current = nul);`. Most of the tests do it, perhaps all tests but the other HttpContextStorageProviderTest. Which would be a bug we could fix right now in the current branch, if you want to.